scripts/generate_pip_deps_from_conda.py

```diff
@@ -87,9 +87,14 @@ def main(conda_fname, pip_fname, compare=False):
         elif isinstance(dep, dict) and len(dep) == 1 and "pip" in dep:
             pip_deps += dep["pip"]
         else:
-            raise ValueError("Unexpected dependency {}".format(dep))
+            raise ValueError(f"Unexpected dependency {dep}")
 
-    pip_content = "\n".join(pip_deps)
+    fname = os.path.split(conda_fname)[1]
+    header = (
+        f"# This file is auto-generated from {fname}, do not modify.\n"
+        "# See that file for comments about the need/usage of each depdendency.\n\n"
+    )
+    pip_content = header + "\n".join(pip_deps)
```

Comment:
Small personal preference, but I think this would be simpler if when defining pip_deps above, instead of `pip_content = []` you write:
```python
pip_content = [
    f"# This file is auto-generated from {fname}, do not modify.",
    "# See that file for comments about the need/usage of each depdendency.",
]
```
If you do that, all this can be deleted here (you'll have to define `fname` before).

```

Reply:
ATM the `pip_deps` list has a specific meaning.  id prefer to retain that

Reply:
Not important, but `headers` could still be a list, and use `"\n".join(headers + pip_deps)`. I think it'll still be an improvement, avoiding the `\n` in the strings. Just saying, happy to get this merge as is.

Reply:
I would actually write this as a triple quoted string.
```
header = f"""
...
..."""

which avoids the embedded newlines

Reply:
triple quoted strings that arent at the module-level mess with code-folding, so i prefer to avoid them

Reply:
k that's what i would have done, but ok as is.